src/couch/src/couch_httpd_auth.erl

```diff
@@ -695,3 +707,12 @@ authentication_warning(#httpd{mochi_req = Req}, User) ->
         "~p: Authentication failed for user ~s from ~s",
         [?MODULE, User, Peer]
     ).
+
+-spec get_config_hash_algorithms() -> list(atom()).
+get_config_hash_algorithms() ->
+    HashAlgorithmStr = string:to_lower(
+        chttpd_util:get_chttpd_auth_config("hash_algorithms", "sha256, sha")
+    ),
+    lists:map(
+        fun binary_to_atom/1, re:split(HashAlgorithmStr, "\\s*,\\s*", [trim, {return, binary}])
```

Comment:
use binary_to_existing_atom so we don't accidentally bloat the atom table with garbage.

Reply:
suggest testing if the item is a member of `crypto:supports(hashs)` instead, and should handle the case where none is valid (suggest in that case we log a warning and default to `sha256`)

Reply:
changed to binary_to_existing_atom in combination with checking the crypto hashes